plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -329,19 +332,142 @@ int CppParser::worker(const clang::tooling::CompileCommand& command_)
 
 CppParser::CppParser(ParserContext& ctx_) : AbstractParser(ctx_)
 {
-  (util::OdbTransaction(_ctx.db))([&, this] {
-    for (const model::BuildAction& ba : _ctx.db->query<model::BuildAction>())
-      _parsedCommandHashes.insert(util::fnvHash(ba.command));
-  });
 }
-  
+
 std::vector<std::string> CppParser::getDependentParsers() const
 {
   return std::vector<std::string>{};
 }
 
+bool CppParser::preparse()
+{
+  std::vector<model::FilePtr> filePtrs(_ctx.fileStatus.size());
+
+  std::transform(_ctx.fileStatus.begin(),
+                 _ctx.fileStatus.end(),
+                 filePtrs.begin(),
+                 [this](const auto& item)
+                 {
+                   if (item.second == IncrementalStatus::MODIFIED)
+                   {
+                     return _ctx.srcMgr.getFile(item.first);
+                   }
+                   else
+                   {
+                     return std::make_shared<model::File>();
+                   }
+                 });
+
+  try
+  {
+    (util::OdbTransaction(_ctx.db))([&]
```

Comment:
Same comment about the vexparse parens, turn them into uniform init.

Reply:
Same response.

Reply:
Fixed in 4301dea623f06401b259435c3999d35a3f646dd6.